go/mcap/error.go

```diff
@@ -0,0 +1,24 @@
+package mcap
```

Comment:
maybe `errors.go`, and move the other errors into this file as well? There are a few error sentinel values lying around: https://pkg.go.dev/github.com/foxglove/mcap/go/mcap#pkg-variables

Reply:
Moved everything into `errors.go`. Fixed up the `UnexpectedTokenError` to be `ErrUnexpectedToken` to match the convention and implemented `error` for the pointer instead.